Sources/Fluent/Entity/KeyNamingConvention.swift

```diff
@@ -0,0 +1,47 @@
+/// The naming convetion to use for foreign
+/// id keys, table names, etc.
+/// ex: snake_case vs. camelCase.
+public enum KeyNamingConvention {
+    case snake_case
+    case camelCase
+}
+
+extension String {
+    internal func snake_case() -> String {
+        let characters = Array(self.characters)
+
+        guard var expanded = characters
+            .first
+            .flatMap({ String($0) })
+            else {
+                return ""
+        }
+
+        characters.suffix(from: 1).forEach { char in
```

Comment:
This one should be fine since we are guarding that we can get a first character.